share/unit/tests/core.sh

```diff
@@ -57,6 +57,12 @@ function testCoreGlobal() {
     #assertEquals "1.4" 2600 $v
 }
 
+function testCoreUnsupportedAssociativeArrayAssignments() {
+    local vetted
+    vetted="$(git grep -E '[a-zA-Z0-9]+\+=\( *\[' | md5sum | cut -d' ' -f1)"
+    assertEquals '0.0.1' '09a2684a0023bdd670ad455efbd74d8e' "${vetted}"
```

Comment:
I don't think it's such a bad idea actually; it's not really unreadable if you know bash - which in many projects is not a safe assumption, but in this project it is :)